spec/models/schema_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Schema, type: :model do
+  context 'when created' do
+    subject { described_class.new }
+
+    it 'must have an endpoint and a structure' do
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+      subject.structure = '{ "field": "value" }'
+      expect { subject.save! }.to raise_error(ActiveRecord::StatementInvalid)
+    end
+
+    it 'must have a non-empty structure' do
+      subject.structure = '{ }'
+      expect { subject.save! }.to raise_error(ActiveRecord::RecordInvalid)
+    end
```

Comment:
Also keep in mind, that it might not really be useful to test ActiveRecord.  Unless you have good reason to do so.